sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobContainerClient.java

```diff
@@ -745,10 +924,39 @@ public void setAccessPolicy(PublicAccessType accessType,
     public Response<Void> setAccessPolicyWithResponse(PublicAccessType accessType,
         List<BlobSignedIdentifier> identifiers, BlobRequestConditions requestConditions,
         Duration timeout, Context context) {
-        Mono<Response<Void>> response = client
-            .setAccessPolicyWithResponse(accessType, identifiers, requestConditions, context);
+        BlobRequestConditions finalRequestConditions = requestConditions == null ? new BlobRequestConditions() : requestConditions;
+
+        if (!ModelHelper.validateNoETag(requestConditions)) {
+            // Throwing is preferred to Mono.error because this will error out immediately instead of waiting until
+            // subscription.
+            throw LOGGER.logExceptionAsError(
+                new UnsupportedOperationException("ETag access conditions are not supported for this API."));
+        }
 
-        return blockWithOptionalTimeout(response, timeout);
+        /*
+        We truncate to seconds because the service only supports nanoseconds or seconds, but doing an
+        OffsetDateTime.now will only give back milliseconds (more precise fields are zeroed and not serialized). This
+        allows for proper serialization with no real detriment to users as sub-second precision on active time for
+        signed identifiers is not really necessary.
+         */
+        if (identifiers != null) {
+            for (BlobSignedIdentifier identifier : identifiers) {
+                if (identifier.getAccessPolicy() != null && identifier.getAccessPolicy().getStartsOn() != null) {
+                    identifier.getAccessPolicy().setStartsOn(
+                        identifier.getAccessPolicy().getStartsOn().truncatedTo(ChronoUnit.SECONDS));
+                }
+                if (identifier.getAccessPolicy() != null && identifier.getAccessPolicy().getExpiresOn() != null) {
+                    identifier.getAccessPolicy().setExpiresOn(
+                        identifier.getAccessPolicy().getExpiresOn().truncatedTo(ChronoUnit.SECONDS));
+                }
+            }
+        }
```

Comment:
This is some fairly complex logic, should add this into `ModelHelper` so that the sync and async clients don't have to define this each.